### ISSUE TITLE
A new trigger type for global control

### ISSUE THREAD

**Issue by namhyung** [namhyung/uftrace#1896]
I remember there were some suggestions using file size or elapsed time as a trigger point and associate (global) actions to it  (But I cannot find where).  For example, we might do things like:
* tracing is off by default, and turn it on after 10 seconds
* similarly, trace only the first 3 seconds of execution
* or trace only from 5 second to 10 second of execution
* finish the tracing when file size reaches to 100 MB

There's `--signal` option for the similar purpose, but it only handles signals.  Maybe we can add time and size support there, or have a new option to handle it separately.

**Comment by honggyukim** [comment]
This is related to #1699.

**Comment by namhyung** [comment]
Actually we can use `timeout` command to send a signal at given time: https://github.com/namhyung/uftrace/discussions/1598#discussioncomment-8188317.

For the first case above, this would do the job (but much more verbose).
```
$ timeout -s RTMIN 10  uftrace --trace=off --signal RTMIN@trace_on <target-program>
```

**Comment by namhyung** [comment]
This could be

```
$ uftrace --trace=off --signal time=10s@trace_on <target-program>
```

**Comment by honggyukim** [comment]
How about the following interface as mentioned at #1699?
```
# trace target program since 1ms elapsed from the beginning.
$ uftrace --trace=10s~ ./prog
```
This can be much simpler.

**Comment by namhyung** [comment]
Yep, that looks better.  I think we also need data size support (maybe only needed for `finish` action).

**Comment by junyeong0619** [comment]
Hello,
I'm currently looking into this issue and would appreciate any guidance. I’ve been looking around some related code, especially in filter.c, and briefly in record.c (though I'm not sure if that's relevant).
Could you please let me know how I might approach understanding or working on this issue? Are there specific areas of the codebase I should pay attention to?

Thank you!

**Comment by namhyung** [comment]
Thanks for your active participation!  The libmcount should handle the trigger, so I think you need to start a timer at `mcount_startup()` and change `mcount_enabled` variable.

**Comment by junyeong0619** [comment]
I've been reviewing the code, and I have a question regarding a potential modification point. For implementing something like uftrace --trace=10s~ ./prog, my initial thought was that we should utilize the existing trace_on and trace_off functionalities within the trigger option. However, I'm unsure if this feature should be introduced as an entirely new trigger or filter option, or if it would be better integrated into the existing trace functionality. Any guidance on this would be greatly appreciated.

**Comment by namhyung** [comment]
For global tracing on-off control, I think it's enough to use the existing `mcount_enabled` variable using signal triggers.  But you need to pick a signal which is not used by the application itself since it'd change the program behavior.  Or you can create a timer internally (in libmcount) and handle that to change the variable.
